cli/cmds_issues.go

```diff
@@ -0,0 +1,153 @@
+package cli
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/katbyte/ghp-pr-sync/lib/gh"
+	"github.com/spf13/cobra"
+	//nolint:misspell
+	c "github.com/gookit/color"
+)
+
+func CmdIssues(_ *cobra.Command, _ []string) error {
+
+	// For each repo get all issues and add to project only bugs
+	//Can't add all issues with current limit on number of issues on a project
+	f := GetFlags()
+	p := gh.NewProject(f.Owner, f.ProjectNumber, f.Token)
+
+	c.Printf("Looking up project details for <green>%s</>/<lightGreen>%d</>...\n", f.Org, f.ProjectNumber)
+	project, err := p.GetProjectDetails()
+	if err != nil {
+		c.Printf("\n\n <red>ERROR!!</> %s", err)
+		return nil
+	}
+	pid := project.Data.Organization.ProjectV2.Id
+	c.Printf("  ID: <magenta>%s</>\n", pid)
+
+	statuses := map[string]string{}
+	fields := map[string]string{}
+
+	for _, f := range project.Data.Organization.ProjectV2.Fields.Nodes {
+		fields[f.Name] = f.Id
+		c.Printf("    <lightBlue>%s</> <> <lightCyan>%s</>\n", f.Name, f.Id)
+
+		if f.Name == "Status" {
+			for _, s := range f.Options {
+				statuses[s.Name] = s.Id
+				c.Printf("      <blue>%s</> <> <cyan>%s</>\n", s.Name, s.Id)
+			}
+		}
+	}
+	fmt.Println()
+	//For issues, only 1 repo will be passed so below will only use f.Repos[0]
+	r := gh.NewRepo(f.Owner, f.Repos[0], f.Token)
+
+	// get all issues
+	c.Printf("Retrieving all issues for <white>%s</>/<cyan>%s</>...", r.Owner, r.Name)
+	issues, err := r.GetAllIssues("open")
+
+	if err != nil {
+		c.Printf("\n\n <red>ERROR!!</> %s\n", err)
+		return nil
+	}
+	c.Printf(" found <yellow>%d</>\n", len(*issues))
+
+	filters := f.GetFilters()
+	fmt.Println("Filtering:")
+
+	//Currently not interested in the username of the author for issues, so I removed the code for now
+	totalIssues := 0
+	daysOpen := 0
+	totalDaysOpen := 0
+
+	for _, issue := range *issues {
+		issueNode := *issue.NodeID
+
+		//only put issues labeled whatever flag is passed (bug, etc) into the project, therefore graphyQL is inside this loop
+
+		for _, f := range filters {
+			match, err := f.Issue(issue)
+			if err != nil {
+				return fmt.Errorf("ERROR: running filter %s: %w", f.Name, err)
+			}
+			if match {
+				c.Printf("Syncing issue <lightCyan>%d</> (<cyan>%s</>) to project.. ", issue.GetNumber(), issueNode)
+				iid, err := p.AddToProject(pid, issueNode)
+				if err != nil {
+					c.Printf("\n\n <red>ERROR!!</> %s", err)
+					continue
+				}
+				c.Printf("<magenta>%s</>", *iid)
+
+				totalIssues++
+				daysOpen = int(time.Now().Sub(issue.GetCreatedAt()) / (time.Hour * 24))
```

Comment:
this is invalid unless you parse events like in prs as it doesn't account for opening/closing/reopening